modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderLocalServiceTest.java

```diff
@@ -0,0 +1,329 @@
+			"Name must only contain letters and digits",
+			() -> _objectFolderLocalService.addObjectFolder(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Abl-e"));
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+		Map<Locale, String> labelMap = LocalizedMapUtil.getLocalizedMap(
+			RandomTestUtil.randomString());
+		String name = RandomTestUtil.randomString();
+
+		ObjectFolder objectFolder = _objectFolderLocalService.addObjectFolder(
+			externalReferenceCode, TestPropsValues.getUserId(), labelMap, name);
+
+		_validateObjectFolder(
+			externalReferenceCode, labelMap, name, objectFolder);
+
+		Assert.assertEquals(
+			1,
+			_resourcePermissionLocalService.getResourcePermissionsCount(
+				objectFolder.getCompanyId(), ObjectFolder.class.getName(),
+				ResourceConstants.SCOPE_INDIVIDUAL,
+				String.valueOf(objectFolder.getObjectFolderId())));
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder);
+	}
+
+	@Test
+	public void testDeleteCompanyObjectFolders() throws Exception {
+		String originalName = PrincipalThreadLocal.getName();
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		Company company = CompanyTestUtil.addCompany();
+
+		ObjectFolder objectFolder = null;
+
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setWithSafeCloseable(
+					company.getCompanyId())) {
+
+			User user = UserTestUtil.getAdminUser(company.getCompanyId());
+
+			Assert.assertNotNull(user);
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(user));
+
+			PrincipalThreadLocal.setName(user.getUserId());
+
+			Assert.assertNotNull(
+				_objectFolderLocalService.getObjectFolder(
+					company.getCompanyId(),
+					ObjectFolderConstants.NAME_UNCATEGORIZED));
+
+			objectFolder = _addObjectFolder(true, user);
+		}
+		finally {
+			_companyLocalService.deleteCompany(company);
+
+			PermissionThreadLocal.setPermissionChecker(
+				originalPermissionChecker);
+			PrincipalThreadLocal.setName(originalName);
+		}
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				company.getCompanyId(),
+				ObjectFolderConstants.NAME_UNCATEGORIZED));
+
+		Assert.assertNull(
+			_objectFolderLocalService.fetchObjectFolder(
+				objectFolder.getObjectFolderId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolder() throws Exception {
+		AssertUtils.assertFailure(
+			UnsupportedOperationException.class,
+			"Uncategorized cannot be deleted",
+			() -> _objectFolderLocalService.deleteObjectFolder(
+				_uncategorizedObjectFolder.getObjectFolderId()));
+
+		ObjectFolder objectFolder = _addObjectFolder(
+			true, TestPropsValues.getUser());
+
+		int countUncategorizedObjectFolder =
+			_objectDefinitionLocalService.getObjectDefinitions(
+				_uncategorizedObjectFolder.getObjectFolderId()
+			).size();
```

Comment:
Do not chain methods: _objectDefinitionLocalService.getObjectDefinitions().size();
Do you have 2 options:

1. Create a method in LocalService: 
```
public int getObjectFolderObjectDefinitionsCount(long objectFolderId) throws PortalException {
		return objectDefinitionPersistence.countByObjectFolderId();
	}
```
2. Declare a variable and get the size;